src/profiles_management/pmr/classes.py

```diff
@@ -0,0 +1,133 @@
+"""This package provides classes for validating and defining a PMR.
+
+The goal of these classes is to represent the PMR in a more
+Pythonic way, rather than a dict of vavlues.
+
+The classes should also provide some common helper methods for
+traversing the information of the PMR, but not functions for
+directly affecting the K8s cluster based on the PMR.
+
+This is similar to how the K8s python clients have classes
+for representing the objects and different methods for doing operations
+with them.
+"""
+
+import logging
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any, Dict, List
+
+import jsonschema
+
+from profiles_management.pmr.schema import RESOURCE_QUOTA_SCHEMA
+
+log = logging.getLogger(__name__)
+
+
+class UserKind(Enum):
+    """Class representing the kind of the user."""
+
+    USER = "user"
+    SERVICE_ACCOUNT = "service-account"
+
+
+class ContributorRole(Enum):
+    """Class representing the role of the user."""
+
+    ADMIN = "admin"
+    EDIT = "edit"
+    VIEW = "view"
+
+
+@dataclass
+class Contributor:
+    """Class representing which users should have access to a Profile."""
+
+    name: str
+    role: ContributorRole
+
+
+@dataclass
+class Owner:
+    """Class representing the owner field of a Profile."""
+
+    name: str
+    kind: UserKind
+
+
+@dataclass
+class Profile:
+    """Class representing a Profile and its Contributors."""
+
+    name: str
+    owner: Owner
+    resources: Dict[str, Any]
+    contributors: List[Contributor]
+
+    # https://docs.python.org/3/library/dataclasses.html#post-init-processing
+    def __post_init__(self):
+        """Validate resourceQuota after values have been initialised."""
+        log.info("Validating ResourceQuota for Profile: %s", self.name)
+        jsonschema.validate(self.resources, RESOURCE_QUOTA_SCHEMA)
```

Comment:
Maybe we want to re-raise here in case of exception so users of this library know what will happen and know what to do when handling the exception.

Reply:
I'm not a big fan of this approach, as it expects users to have to go and look at the source code.

I like the idea of exposing that a `ValidationError` will be raised in the docstring though, so it's shown in devs' editors

Reply:
Pushed a commit. Although looking at it again, since the docstring is in `__post_init__` I don't think anyone will manage to see this.

Still, I prefer this over having code for exposing the error
b679497

Reply:
Yep! I've done this in f170187 WDYT?